src/commands/mobile_app/upload.rs

```diff
@@ -198,19 +198,106 @@ fn validate_is_mobile_app(path: &Path, bytes: &[u8]) -> Result<()> {
     ))
 }
 
+fn ipa_to_xcarchive(ipa_path: &Path, ipa_bytes: &[u8]) -> Result<TempFile> {
+    debug!(
+        "Converting IPA to XCArchive structure: {}",
+        ipa_path.display()
+    );
+
+    let temp_dir = crate::utils::fs::TempDir::create()?;
+    let xcarchive_dir = temp_dir.path().join("archive.xcarchive");
+    let products_dir = xcarchive_dir.join("Products");
+    let applications_dir = products_dir.join("Applications");
+
+    debug!("Creating XCArchive directory structure");
+    std::fs::create_dir_all(&applications_dir)?;
+
+    // Extract IPA file
+    let cursor = std::io::Cursor::new(ipa_bytes);
+    let mut ipa_archive = zip::ZipArchive::new(cursor)?;
+
+    let mut app_name = String::new();
+
+    // Extract .app from Payload/ directory
+    for i in 0..ipa_archive.len() {
+        let mut file = ipa_archive.by_index(i)?;
+        let file_path = file.name().to_string();
+
+        if let Some(stripped) = file_path.strip_prefix("Payload/") {
+            if let Some(app_folder_name) = stripped.strip_suffix(".app/") {
+                app_name = app_folder_name.to_string();
+                debug!("Found app: {}", app_name);
+            }
+
+            if !file.is_dir() {
+                // Create the file path in the XCArchive structure
+                let target_path = applications_dir.join(stripped);
+
+                // Create parent directories
+                if let Some(parent) = target_path.parent() {
+                    std::fs::create_dir_all(parent)?;
+                }
+
+                // Extract file
+                let mut target_file = std::fs::File::create(&target_path)?;
+                std::io::copy(&mut file, &mut target_file)?;
+            }
+        }
+    }
+
+    // Create Info.plist for XCArchive
+    let info_plist_path = xcarchive_dir.join("Info.plist");
+
+    let info_plist_content = format!(
+        r#"<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>ApplicationProperties</key>
+	<dict>
+		<key>ApplicationPath</key>
+		<string>Applications/{app_name}.app</string>
+	</dict>
+	<key>ArchiveVersion</key>
+	<integer>1</integer>
+</dict>
+</plist>"#
+    );
+
+    std::fs::write(&info_plist_path, info_plist_content)?;
+
+    debug!(
+        "Created XCArchive Info.plist at: {}",
+        info_plist_path.display()
+    );
+    normalize_directory(&xcarchive_dir)
+}
+
 // For APK and AAB files, we'll copy them directly into the zip
+// For IPA files, we'll convert them to XCArchive structure first
 fn normalize_file(path: &Path, bytes: &[u8]) -> Result<TempFile> {
     debug!("Creating normalized zip for file: {}", path.display());
 
+    // Check if this is an IPA file that needs conversion
+    if is_zip_file(bytes) && is_ipa_file(bytes)? {
+        debug!("Converting IPA file to XCArchive structure");
+        return ipa_to_xcarchive(path, bytes);
+    }
```

Comment:
I think it would make more sense to leave this function unchanged, and instead check whether we are dealing with an IPA file prior to calling `normalize_file`